src/App.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('click', this.handleRightClick);
+    document.addEventListener('contextmenu', this.handleLeftClick);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidUpdate(_: {}, prevState: State) {
+    const { clockName, hasClock } = this.state;
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    if (clockName !== prevState.clockName && hasClock) {
+      window.console.debug(
+        `Renamed from ${prevState.clockName} to ${clockName}`,
+      );
+    }
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    clearInterval(this.timerId);
+  }
```

Comment:
You should cancel your eventListeners too if you unmounting component
Strings 28-29